tools/resample/resample.hpp

```diff
@@ -0,0 +1,170 @@
+        {
+            std::cout << "\nUsage: resample [-h] [-v] <input> <grid> <output>" << std::endl;
+            std::cout << "-v: toggle verbose messages (now=" << verbose << ")" << std::endl;
+        }
+
+        void parse_param(int argc, char *argv[]) 
+        {
+            char c;
+            while (c = getopt(argc, argv, "hvp:"), c != -1) 
+            {
+                switch (c) 
+                {
+                case 'h':
+                    usage();
+                    exit(EXIT_SUCCESS);
+                case 'v': verbose = true; break;
+                case 'p': output_precision = atoi(optarg); break;
+                default: throw std::runtime_error("Unknown argument " + c);
+                }
+            }
+            int remaining = argc - optind;
+            if (remaining != 3) {
+                about();
+                usage();
+                exit(1);
+            }
+            inputfn  = std::string(argv[optind++]);
+            gridfn   = std::string(argv[optind++]);
+            outputfn = std::string(argv[optind++]);
+        }
+
+        void about() 
+        {
+            std::cout << "Resampling tool" << std::endl;
+            #ifdef __TIMESTAMP__
+            std::cout << "Timestamp: " << __TIMESTAMP__ << std::endl;
+            #endif
+            std::cout << "Compiled on " << __DATE__ << " at " << __TIME__ << std::endl;
+        }
+             
+
+    public:
+        Resample(int argv, char *argc[])
+        {
+            parse_param(argv, argc);
+            if (verbose) about();
+        }
+
+        Resample(std::string inputfn, std::string gridfn, std::string outputfn, bool verbose = false, int output_precision = 16){
+            this->inputfn = inputfn;
+            this->gridfn = gridfn;
+            this->outputfn = outputfn;
+            this->verbose = verbose;
+            this->output_precision = output_precision;
+            if (verbose) about();
+        }
+        
+        ~Resample() 
+        {
+            gsl_spline_free(spline);
+            gsl_interp_accel_free(acc);
+        }
+
+        void run()
+        {
+            auto Fin = safe_open_for_reading(inputfn);
+            std::vector<std::pair<double, double>> f;
+            readtable(Fin, f, verbose);
+            init(f);
+
+            auto Fgrid = safe_open_for_reading(gridfn);
+            std::vector<std::pair<double, double>> grid;
+            readtable(Fgrid, grid, verbose);
+            resample(grid);
+
+            auto Fout = safe_open(outputfn);
+            writetable(grid, Fout, output_precision);
+        }
+
+        void init(std::vector<std::pair<T, T>> &im) 
+        {
+            std::sort(im.begin(), im.end());
+            len  = im.size();
+            Xmin = im[0].first;
+            Xmax = im[len - 1].first;
+            if (verbose) std::cout << "Range: [" << Xmin << " ; " << Xmax << "]" << std::endl;
+            // Xpts are increasing
+            Xpts = std::vector<T> (len);
```

Comment:
To bi slo z dvema klicema std::copy namesto s for zanko.

Reply:
Probavam zdaj, ampak ne vem kako bi to naredil

Reply:
std::copy(im.begin(), im.end(), Xpts.begin(), [](const auto &pair) { return pair.first; });

Druga moznost je ze napisana funkcija split_vector_of_pairs v kk.hpp.

Najboljsa moznost pa je, da uporabis za napisani class interpolator iz hilb.hpp (ki bi ga bilo smiselno preseliti, skupaj z class integrator, v c++/numerics.hpp). V tem primeru se bo resample se bolj poenostavil, pa tudi za memory management je ze poskrbljeno.